backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -66,17 +66,24 @@ class BibTexCommonSchema(Schema):
 
     @staticmethod
     def get_date(data, doc_type):
+        from inspirehep.records.api import LiteratureRecord
+
         publication_year = BibTexCommonSchema.get_best_publication_info(data).get(
             "year"
         )
         thesis_date = get_value(data, "thesis_info.date")
         imprint_date = get_value(data, "imprints.date[0]")
-
-        if doc_type.endswith("thesis"):
-            date_choice = thesis_date or publication_year or imprint_date
+        earliest_date = LiteratureRecord(data).earliest_date
+        date_map = {
+            "phdthesis": thesis_date,
+            "masterthesis": thesis_date,
+            "book": imprint_date,
+            "book chapter": imprint_date,
```

Comment:
are these really possible `doc_type`s?

Reply:
Yes, since we take them from data, the same case as with
`document_type_map = {
"article": "article",  
"book": "book",  
"book chapter": "inbook"}`

Reply:
but `phdthesis` and `masterthesis` are not INSPIRE document types. So either the first two are wrong, or the last one is.